frontend/online-shop/src/components/Header.tsx

```diff
@@ -0,0 +1,40 @@
+import { AppBar, Button, Toolbar, Typography } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import ShopIcon from "@material-ui/icons/Shop";
+import { Link } from "react-router-dom";
+
+const useStyles = makeStyles((theme) => ({
```

Comment:
Those styles should be extracted into a different file called `Header.styles.tsx` and imported from there. Also if you want to use the theme properties you have to also call `createStyles` within the anonymous method and give theme the `Theme` type.